contrib/drivers/filesharedrivers/oceanstor/oceanstor.go

```diff
@@ -473,34 +474,26 @@ func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) er
 }
 
 func (d *Driver) getAccessLevel(accessLevels []string, shareProto string) (string, error) {
+	var accessLevel string
+
 	if accessLevels == nil || (accessLevels != nil && len(accessLevels) == 0) {
 		return "", errors.New("access level cannot be empty")
 	}
 
-	if len(accessLevels) == 1 && strings.ToLower(accessLevels[0]) != AccessLevelRead {
-		return "", errors.New("only read only and read write access level are supported")
+	if len(accessLevels) > 2 {
+		return "", errors.New("invalid access level")
 	}
 
-	var accessLevel string
-	// make sure accessLevel is ro or rw
-	for _, level := range accessLevels {
-		if strings.ToLower(level) == AccessLevelExecute {
-			return "", errors.New("only read only and read write access level are supported, execute is not supported")
-		}
-		if strings.ToLower(level) == AccessLevelRead && !strings.Contains(accessLevel, "r") {
-			accessLevel += "r"
-		}
-	}
-
-	for _, level := range accessLevels {
-		if strings.ToLower(level) == AccessLevelWrite {
-			accessLevel += "w"
-			break
-		}
+	if len(accessLevels) == 1 && strings.ToLower(accessLevels[0]) != AccessLevelRead {
+		return "", errors.New("only read only or read write access level are supported")
+	} else {
+		accessLevel = "ro"
 	}
 
-	if accessLevel == "r" {
-		accessLevel += "o"
+	if !utils.Contained(AccessLevelRead, accessLevels) || !utils.Contained(AccessLevelWrite, accessLevels) {
+		return "", errors.New("only read only or read write access level are supported")
+	} else {
+		accessLevel = "rw"
```

Comment:
the conditional codes is not right

Reply:
only write case will pass? please check the conditional code.